src/v/cluster/partition_balancer_backend.cc

```diff
@@ -358,9 +364,22 @@ ss::future<> partition_balancer_backend::do_tick() {
     // status requests by default 700ms
     const auto node_responsiveness_timeout = _node_status_interval() * 7;
 
+    const bool should_sanction = _feature_table.should_sanction();
+
+    const auto [mode, is_sanctioned] = _mode(should_sanction);
+    if (is_sanctioned) {
+        vlog(
+          clusterlog.warn,
+          "A Redpanda Enterprise Edition license is required to use enterprise "
+          "feature \"partition_autobalancing_mode\" with value \"{}\". "
+          "Behavior is being restricted to \"{}\".",
```

Comment:
I really like this last sentence you've added, it's super useful to include what we're changing the value to.